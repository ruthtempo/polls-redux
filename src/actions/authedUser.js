export const SET_AUTHED_USER = "SET_AUTHED_USER";

export default function setAuthedUser(userId) {
  return {
    type: SET_AUTHED_USER,
    userId,
  };
}
