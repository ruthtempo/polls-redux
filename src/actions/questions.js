import * as API from "../_DATA";
import { showLoading, hideLoading } from "react-redux-loading-bar";

export const RECEIVE_QUESTIONS_DATA = "RECEIVE_QUESTIONS_DATA";

const receiveQuestions = (questions) => {
  return {
    type: RECEIVE_QUESTIONS_DATA,
    questions,
  };
};

export const handleInitialQuestionsData = () => {
  return (dispatch) => {
    dispatch(showLoading());
    return API._getQuestions().then((questions) => {
      dispatch(hideLoading());
      dispatch(receiveQuestions(questions));
    });
  };
};
