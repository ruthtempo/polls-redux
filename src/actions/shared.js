import API from "../_DATA";

export const RECEIVE_DATA = "RECEIVE_DATA";

const receiveData = (users, questions) => {
  return {
    type: RECEIVE_DATA,
    users,
    questions,
  };
};

export const handleInitialData = () => {
  return (dispatch) => {
    return Promise.all([API._getUsers(), API._getQuestions()]).then(
      ([users, questions]) => {
        dispatch(receiveData(users, questions));
      }
    );
  };
};
