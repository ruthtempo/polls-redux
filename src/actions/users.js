import * as API from "../_DATA";

export const RECEIVE_USERS_DATA = "RECEIVE_USERS_DATA";

export const receiveData = (users) => {
  return {
    type: RECEIVE_USERS_DATA,
    users,
  };
};

export const handleInitialUsersData = () => {
  return (dispatch) => {
    return API._getUsers().then((users) => {
      dispatch(receiveData(users));
    });
  };
};
