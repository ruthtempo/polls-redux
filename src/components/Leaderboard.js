import { connect } from "react-redux";

const Leaderboard = ({ users }) => {
  return (
    users && (
      <div className="flex justify-center ">
        <table className="table-auto bg-indigo-200lg:w-2/3 2xl:w-1/3 xl:w-1/3 md:w-2/3">
          <thead className="h-11">
            <tr className="bg-indigo-400 text-white">
              <th>Users</th>
              <th>Answered</th>
              <th>Created</th>
            </tr>
          </thead>
          <tbody>
            {Object.values(users).map((user) => (
              <tr key={user.id}>
                <td className="flex flex-col md:flex-row p-6 items-center">
                  <img
                    src={user.avatarURL}
                    alt="user-avatar"
                    className="object-contain rounded-full mr-3 w-9 lg:w-20"
                  />
                  <p>{user.id}</p>
                </td>
                <td className="text-end pr-6">
                  {Object.values(user.answers).length}
                </td>
                <td className="text-end pr-6">
                  {Object.values(user.questions).length}
                </td>
              </tr>
            ))}
          </tbody>
        </table>
      </div>
    )
  );
};

const mapStateToProps = ({ users }) => {
  const usersWithTotalpolls = Object.values(users).map((user) => ({
    ...user,
    totalPoints:
      Object.values(user.questions).length + Object.values(user.answers).length,
  }));

  const sortedUsers = usersWithTotalpolls.sort(
    (a, b) => b.totalPoints - a.totalPoints
  );
  return {
    users: sortedUsers,
  };
};

export default connect(mapStateToProps)(Leaderboard);
