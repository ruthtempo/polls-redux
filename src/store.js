import middleware from "./middleware";
import reducer from "./reducers";
import { createStore } from "redux";

export const store = createStore(reducer, middleware);
