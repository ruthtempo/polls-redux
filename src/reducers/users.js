import { RECEIVE_USERS_DATA } from "../actions/users";

export default function users(state = null, action) {
  switch (action.type) {
    case RECEIVE_USERS_DATA:
      return action.users;
  }
  return state;
}
