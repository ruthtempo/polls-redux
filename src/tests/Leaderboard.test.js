import { store } from "../store";
import { Provider } from "react-redux";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Leaderboard from "../components/Leaderboard";
import { receiveData } from "../actions/users";
import { users } from "../_DATA";

describe("leaderboard", () => {
  it("will match snapshot", () => {
    store.dispatch(receiveData(users));
    const component = render(
      <Provider store={store}>
        <MemoryRouter>
          <Leaderboard />
        </MemoryRouter>
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
});
