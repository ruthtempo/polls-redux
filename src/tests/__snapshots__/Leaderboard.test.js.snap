// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`leaderboard will match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h3
          class="text-center text-3xl py-3"
        >
          Leaderboard
        </h3>
        <div
          class="flex justify-center hidden md:flex"
        >
          <table
            class="table-auto bg-indigo-200 md:w-2/3 xl:w-1/2 2xl:w-1/3 rounded-lg border-separate "
          >
            <thead
              class="h-11 bg-indigo-400 text-white border-md"
            >
              <tr>
                <th>
                  Users
                </th>
                <th>
                  Answered
                </th>
                <th>
                  Created
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td
                  class="flex flex-col md:flex-row p-6 items-center"
                >
                  <img
                    alt="user-avatar"
                    class="object-contain rounded-full mr-3 w-9 lg:w-20"
                    src="sarah.jpg"
                  />
                  <p>
                    sarahedo
                  </p>
                </td>
                <td
                  class="text-end pr-6"
                >
                  4
                </td>
                <td
                  class="text-end pr-6"
                >
                  2
                </td>
              </tr>
              <tr>
                <td
                  class="flex flex-col md:flex-row p-6 items-center"
                >
                  <img
                    alt="user-avatar"
                    class="object-contain rounded-full mr-3 w-9 lg:w-20"
                    src="mike.jpg"
                  />
                  <p>
                    mtsamis
                  </p>
                </td>
                <td
                  class="text-end pr-6"
                >
                  3
                </td>
                <td
                  class="text-end pr-6"
                >
                  2
                </td>
              </tr>
              <tr>
                <td
                  class="flex flex-col md:flex-row p-6 items-center"
                >
                  <img
                    alt="user-avatar"
                    class="object-contain rounded-full mr-3 w-9 lg:w-20"
                    src="tyler.jpg"
                  />
                  <p>
                    tylermcginnis
                  </p>
                </td>
                <td
                  class="text-end pr-6"
                >
                  2
                </td>
                <td
                  class="text-end pr-6"
                >
                  2
                </td>
              </tr>
              <tr>
                <td
                  class="flex flex-col md:flex-row p-6 items-center"
                >
                  <img
                    alt="user-avatar"
                    class="object-contain rounded-full mr-3 w-9 lg:w-20"
                    src="zenobi.jpg"
                  />
                  <p>
                    zoshikanlu
                  </p>
                </td>
                <td
                  class="text-end pr-6"
                >
                  1
                </td>
                <td
                  class="text-end pr-6"
                >
                  0
                </td>
              </tr>
            </tbody>
          </table>
        </div>
        <div
          class="md:hidden mx-2 my-4 shadow-lg rounded-lg bg-indigo-200 flex flex-col py-2 border-teal-400 border-2"
        >
          <div
            class="flex items-center justify-center p-4"
          >
            <img
              alt="user avatar"
              class="rounded-full w-9"
              src="sarah.jpg"
            />
            <div
              class="mx-2"
            >
              sarahedo
            </div>
          </div>
          <div
            class="flex items-center justify-between p-4"
          >
            <div>
              Answers
            </div>
            <div>
              4
            </div>
          </div>
          <div
            class="flex items-center justify-between p-4 "
          >
            <div>
              Created
            </div>
            <div>
              2
            </div>
          </div>
        </div>
        <div
          class="md:hidden mx-2 my-4 shadow-lg rounded-lg bg-indigo-200 flex flex-col py-2 border-teal-400 border-2"
        >
          <div
            class="flex items-center justify-center p-4"
          >
            <img
              alt="user avatar"
              class="rounded-full w-9"
              src="mike.jpg"
            />
            <div
              class="mx-2"
            >
              mtsamis
            </div>
          </div>
          <div
            class="flex items-center justify-between p-4"
          >
            <div>
              Answers
            </div>
            <div>
              3
            </div>
          </div>
          <div
            class="flex items-center justify-between p-4 "
          >
            <div>
              Created
            </div>
            <div>
              2
            </div>
          </div>
        </div>
        <div
          class="md:hidden mx-2 my-4 shadow-lg rounded-lg bg-indigo-200 flex flex-col py-2 border-teal-400 border-2"
        >
          <div
            class="flex items-center justify-center p-4"
          >
            <img
              alt="user avatar"
              class="rounded-full w-9"
              src="tyler.jpg"
            />
            <div
              class="mx-2"
            >
              tylermcginnis
            </div>
          </div>
          <div
            class="flex items-center justify-between p-4"
          >
            <div>
              Answers
            </div>
            <div>
              2
            </div>
          </div>
          <div
            class="flex items-center justify-between p-4 "
          >
            <div>
              Created
            </div>
            <div>
              2
            </div>
          </div>
        </div>
        <div
          class="md:hidden mx-2 my-4 shadow-lg rounded-lg bg-indigo-200 flex flex-col py-2 border-teal-400 border-2"
        >
          <div
            class="flex items-center justify-center p-4"
          >
            <img
              alt="user avatar"
              class="rounded-full w-9"
              src="zenobi.jpg"
            />
            <div
              class="mx-2"
            >
              zoshikanlu
            </div>
          </div>
          <div
            class="flex items-center justify-between p-4"
          >
            <div>
              Answers
            </div>
            <div>
              1
            </div>
          </div>
          <div
            class="flex items-center justify-between p-4 "
          >
            <div>
              Created
            </div>
            <div>
              0
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h3
        class="text-center text-3xl py-3"
      >
        Leaderboard
      </h3>
      <div
        class="flex justify-center hidden md:flex"
      >
        <table
          class="table-auto bg-indigo-200 md:w-2/3 xl:w-1/2 2xl:w-1/3 rounded-lg border-separate "
        >
          <thead
            class="h-11 bg-indigo-400 text-white border-md"
          >
            <tr>
              <th>
                Users
              </th>
              <th>
                Answered
              </th>
              <th>
                Created
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td
                class="flex flex-col md:flex-row p-6 items-center"
              >
                <img
                  alt="user-avatar"
                  class="object-contain rounded-full mr-3 w-9 lg:w-20"
                  src="sarah.jpg"
                />
                <p>
                  sarahedo
                </p>
              </td>
              <td
                class="text-end pr-6"
              >
                4
              </td>
              <td
                class="text-end pr-6"
              >
                2
              </td>
            </tr>
            <tr>
              <td
                class="flex flex-col md:flex-row p-6 items-center"
              >
                <img
                  alt="user-avatar"
                  class="object-contain rounded-full mr-3 w-9 lg:w-20"
                  src="mike.jpg"
                />
                <p>
                  mtsamis
                </p>
              </td>
              <td
                class="text-end pr-6"
              >
                3
              </td>
              <td
                class="text-end pr-6"
              >
                2
              </td>
            </tr>
            <tr>
              <td
                class="flex flex-col md:flex-row p-6 items-center"
              >
                <img
                  alt="user-avatar"
                  class="object-contain rounded-full mr-3 w-9 lg:w-20"
                  src="tyler.jpg"
                />
                <p>
                  tylermcginnis
                </p>
              </td>
              <td
                class="text-end pr-6"
              >
                2
              </td>
              <td
                class="text-end pr-6"
              >
                2
              </td>
            </tr>
            <tr>
              <td
                class="flex flex-col md:flex-row p-6 items-center"
              >
                <img
                  alt="user-avatar"
                  class="object-contain rounded-full mr-3 w-9 lg:w-20"
                  src="zenobi.jpg"
                />
                <p>
                  zoshikanlu
                </p>
              </td>
              <td
                class="text-end pr-6"
              >
                1
              </td>
              <td
                class="text-end pr-6"
              >
                0
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        class="md:hidden mx-2 my-4 shadow-lg rounded-lg bg-indigo-200 flex flex-col py-2 border-teal-400 border-2"
      >
        <div
          class="flex items-center justify-center p-4"
        >
          <img
            alt="user avatar"
            class="rounded-full w-9"
            src="sarah.jpg"
          />
          <div
            class="mx-2"
          >
            sarahedo
          </div>
        </div>
        <div
          class="flex items-center justify-between p-4"
        >
          <div>
            Answers
          </div>
          <div>
            4
          </div>
        </div>
        <div
          class="flex items-center justify-between p-4 "
        >
          <div>
            Created
          </div>
          <div>
            2
          </div>
        </div>
      </div>
      <div
        class="md:hidden mx-2 my-4 shadow-lg rounded-lg bg-indigo-200 flex flex-col py-2 border-teal-400 border-2"
      >
        <div
          class="flex items-center justify-center p-4"
        >
          <img
            alt="user avatar"
            class="rounded-full w-9"
            src="mike.jpg"
          />
          <div
            class="mx-2"
          >
            mtsamis
          </div>
        </div>
        <div
          class="flex items-center justify-between p-4"
        >
          <div>
            Answers
          </div>
          <div>
            3
          </div>
        </div>
        <div
          class="flex items-center justify-between p-4 "
        >
          <div>
            Created
          </div>
          <div>
            2
          </div>
        </div>
      </div>
      <div
        class="md:hidden mx-2 my-4 shadow-lg rounded-lg bg-indigo-200 flex flex-col py-2 border-teal-400 border-2"
      >
        <div
          class="flex items-center justify-center p-4"
        >
          <img
            alt="user avatar"
            class="rounded-full w-9"
            src="tyler.jpg"
          />
          <div
            class="mx-2"
          >
            tylermcginnis
          </div>
        </div>
        <div
          class="flex items-center justify-between p-4"
        >
          <div>
            Answers
          </div>
          <div>
            2
          </div>
        </div>
        <div
          class="flex items-center justify-between p-4 "
        >
          <div>
            Created
          </div>
          <div>
            2
          </div>
        </div>
      </div>
      <div
        class="md:hidden mx-2 my-4 shadow-lg rounded-lg bg-indigo-200 flex flex-col py-2 border-teal-400 border-2"
      >
        <div
          class="flex items-center justify-center p-4"
        >
          <img
            alt="user avatar"
            class="rounded-full w-9"
            src="zenobi.jpg"
          />
          <div
            class="mx-2"
          >
            zoshikanlu
          </div>
        </div>
        <div
          class="flex items-center justify-between p-4"
        >
          <div>
            Answers
          </div>
          <div>
            1
          </div>
        </div>
        <div
          class="flex items-center justify-between p-4 "
        >
          <div>
            Created
          </div>
          <div>
            0
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
